src/clients/PuchasesClient.ts

```diff
@@ -0,0 +1,86 @@
+import { IPurchasesClient } from "../interfaces/purchases.interfaces";
+import { ApiCustomerPurchases, ApiCustomerSummary, ClientOptions, GetCustomerPurchasesCallback, GetCustomersBulkInfoCallback, GetCustomersBulkInfoParams, Paginated } from "../types";
+import { BaseClient } from "./BaseClient.abstract";
+
+/**
+ * class PurchasesClient
+ * constructor requires options as
+ * {
+ *  protocol: 'http',
+ *  hostname: 'domain.com',
+ *  port: 80,
+ *  pathnamePrefix?: '/v2/customers',
+ *  clientId?: 'my-client-id',
+ *  secret?: 'secret key if exists',
+ *  agent?: string
+ * }
+ */
+export class PurchasesClient extends BaseClient implements IPurchasesClient {
+
+  /**
+   * constructor
+   * @param options ClientOptions
+   */
+  constructor(options: ClientOptions) {
+    super(`${options.protocol}://${options.hostname}:${options.port}${options.pathnamePrefix}`, options);
+
```

Comment:
So you advise here to rename it to CustomersClientV2 => 
and then it will use the prefixname /v2/customers.
The thing is what happends if there is /v3/customers ? do we change the class name ? or we add another class CustomersClientV3 ?

and what about the undocumented endpoints .. ?
from where we can get the details about them ?
what if we can now finish the required endpoints to be used for the purchases-subscription, then we can add the remaining endpoints ?

Reply:
Later on we might add a "CustomersClientV3" (and keep the V2). For example when writing API v2 in the server, I wrote a compatibility layer so `v1` endpoints still work. Apps using the `V1` would continue to work (but with an inferior API that can't handle certain edge cases).

Forget about undocumented endpoint for now, nothing urgent. We'll probably add them in the future though.

I can demo them when we talk if you want to get an idea.

Reply:
`v2.ts` exports the V2 classes. Prepare code architecture for v3 (a v2 folder?).